### PR TITLE
Ensure YamlCreate re-encodes spaces

### PR DESCRIPTION
**Pull request by Trenly**
cc @mdanish-kh
 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/113795&drop=dogfoodAlpha

### CODE REVIEW

**Review by mdanish-kh — COMMENTED** [review]
We might want to replace the whitespace after the file has been downloaded. I tried with `https://edu-wenku.bdimg.com/v1/pcclient/upgrade/百度文库 Setup 1.2.7.exe` from manifests/b/Baidu/BaiduWenku/1.2.7 and this change seems to be breaking for it.

Edit: Thinking more.. I'm not sure how the pipelines will act with these URLs if they break here in the script.

**Comment by Trenly** [comment]
> https://edu-wenku.bdimg.com/v1/pcclient/upgrade/百度文库 Setup 1.2.7.exe

Interesting; I tried that URL manually in a browser and it worked fine for me, although the browser seems to have decoded the non-english characters

**Comment by mdanish-kh** [comment]
Interestingly, even the decoded URL doesn't seem to work

image

**Comment by Trenly** [comment]
> Interestingly, even the decoded URL doesn't seem to work
> 
> image

Might be an issue with powershell then? Seemed to work fine with other URLs I tried

**Review by stephengillie — APPROVED** [review]

File: Tools/YamlCreate.ps1 [hunk 1]
```diff
@@ -454,6 +454,7 @@ Function Request-InstallerUrl {
         }
       }
       $NewInstallerUrl = [System.Web.HttpUtility]::UrlDecode($NewInstallerUrl.Replace('+','%2B'))
+      $NewInstallerUrl.Replace(' ','%20')
```
Comment (SpecterShell) [inline Tools/YamlCreate.ps1:457]:
```
$NewInstallerUrl = $NewInstallerUrl.Replace(' ','%20')
```

**Pull request merged**